feelslikemonday/app/src/androidTest/java/com/example/feelslikemonday/DAO/UserDAOTest.java

```diff
@@ -60,7 +60,107 @@ public void onCallback() {
+        final CountDownLatch signal = new CountDownLatch(1);
+        userDAO.createOrUpdate(new User("uTEST-will", "password"), new VoidCallback() {
+            @Override
+            public void onCallback() {
+                userDAO.delete(new User("uTEST-will","password"), new VoidCallback() {
+                    @Override
+                    public void onCallback() {
+                        userDAO.get("uTEST-will", new UserCallback() {
+                            @Override
+                            public void onCallback(User user) {
+                                fail();
+                                signal.countDown();
+                            }
+                        }, new VoidCallback() {
+                            @Override
+                            public void onCallback() {
+                                //No user was found
+                                signal.countDown();
+                            }
+                        });
+                    }
+                });
+            }
+        });
+        signal.await();
+    }
 
+    //Create a user, check if it exists, delete the user, check if it exists again
+    @Test
+    public void checkIfUserExists() throws InterruptedException{
+        final CountDownLatch signal = new CountDownLatch(1);
+        userDAO.createOrUpdate(new User("uTEST-nill", "password"), new VoidCallback() {
+            @Override
+            public void onCallback() {
+                userDAO.checkIfExists("uTEST-nill", new BooleanCallback() {
+                    @Override
+                    public void onCallback(Boolean bool) {
+                        assertEquals(bool, Boolean.TRUE);
+                        userDAO.delete(new User("uTEST-nill", "password"), new VoidCallback() {
+                            @Override
+                            public void onCallback() {
+                                userDAO.checkIfExists("uTEST-nill", new BooleanCallback() {
+                                    @Override
+                                    public void onCallback(Boolean bool) {
+                                        assertEquals(bool, Boolean.FALSE);
+                                        signal.countDown();
+                                    }
+                                }, new VoidCallback() {
+                                    @Override
+                                    public void onCallback() {
+                                        fail();
+                                        signal.countDown();
+                                    }
+                                });
+                            }
+                        });
+                    }
+                }, new VoidCallback() {
+                    @Override
+                    public void onCallback() {
+                        fail();
+                        signal.countDown();
+                    }
+                });
+            }
+        });
```

Comment:
also need a test just to see if a user is made when we add a new user / sign up not just for updating

Reply:
I would delegate this test to the log-in/sign-in activity test. We have a test that creates a user using the DAO alone. The tests you have mentioned are definitely things we should be testing for though.